packages/CodeGenLib/src/Database/sql_codegen.ts

```diff
@@ -675,6 +694,28 @@ export class SQLCodeGenBase {
 `
     }
 
+    /**
+     * Generates the SQL for creating indexes for the foreign keys in the entity
+     * @param ds 
+     * @param entity 
+     */
+    generateIndexesForForeignKeys(ds: DataSource, entity: EntityInfo): string {
+        // iterate through all of the fields in the entity that are foreign keys and generate an index for each one
+        let sOutput: string = ''; 
+        for (const f of entity.Fields) {
+            if (f.RelatedEntity && f.RelatedEntity.length > 0) {
```

Comment:
Zero-length strings are falsy, so the `&& f.RelatedEntity.length > 0` will never get evaluated

Reply:
Thanks for that, we can optimize the readability of code in many areas with that insight. The optimizer probably eliminates that anyway in transpilation but even if it ends up in JS as you said the eval short circuits so it has no perf overhead beyond parsing, but good to get rid of IMO as much for simplifying readability for us humans :)